packages/flutter_tools/lib/src/dart/dependencies.dart

```diff
@@ -37,7 +37,8 @@ class DartDependencySetBuilder {
   final Uri _mainScriptUri;
 
   Set<String> build() {
-    final List<String> dependencies = <String>[_mainScriptPath, _packagesFilePath];
+    final List<String> dependencies = <String>[_mainScriptPath,
+        _mainScriptPath + ".dill", _packagesFilePath];
```

Comment:
Good catch, thanks! Actually I think it's altogether unnecessary to track .dill file here. It will be produced and added to the dirtyEntries list in devfs.dart update() directly if any source files were changed.